src/ray/util/stream_redirection.cc

```diff
@@ -36,12 +39,27 @@ namespace {
 absl::flat_hash_map<MEMFD_TYPE_NON_UNIQUE, internal::StreamRedirectionHandle>
     redirection_file_handles;
 
+// A validation function, which verifies redirection handles don't dump to the same file.
+void ValidateOutputPathsUnique() {
+  absl::InlinedVector<std::string_view, 2> filepaths;
+  for (const auto &[_, handle] : redirection_file_handles) {
+    const auto &cur_filepath = handle.GetFilePath();
+    // Skip empty filepaths, which means current handle doesn't redirect and persist.
+    if (!cur_filepath.empty()) {
```

Comment:
I don't think it can be empty, it should be a check instead?

Reply:
What if we don't redirect to file?

Reply:
ok